File: mobile/lib/core/vocabulary.dart

class FiCategory {
  const FiCategory(this.label, this.series);

  final String label;
  final int series;
}

const Map<String, FiCategory> fiCategories = {
  'renda_fixa': FiCategory('Renda Fixa', 1),
  'acoes_br': FiCategory('Ações BR', 2),
  'fiis': FiCategory('FIIs', 3),
  'bdrs': FiCategory('BDRs', 5),
  'etfs': FiCategory('ETFs', 8),
  'auto': FiCategory('Automática', 0),
};

const Map<String, String> fiCategoryAliases = {
  'renda': 'renda_fixa',
  'caixa': 'renda_fixa',
  'trade': 'acoes_br',
};

const Map<String, String> fiAssetTypes = {
  'br_stock': 'Ação BR',
  'bdr': 'BDR',
  'fii': 'FII',
  'etf': 'ETF',
  'renda_fixa': 'Renda Fixa',
};

const Map<String, String> fiAssetTypeToCategory = {
  'br_stock': 'acoes_br',
  'bdr': 'bdrs',
  'fii': 'fiis',
  'etf': 'etfs',
  'renda_fixa': 'renda_fixa',
};

const Map<String, String> fiSectors = {
  'Financial Services': 'Financeiro',
  'Technology': 'Tecnologia',
  'Energy': 'Energia',
  'Consumer Cyclical': 'Consumo Cíclico',
  'Healthcare': 'Saúde',
  'Industrials': 'Industrial',
  'Real Estate': 'Imobiliário',
  'Consumer Defensive': 'Consumo Básico',
  'Basic Materials': 'Materiais Básicos',
  'Utilities': 'Utilidades Públicas',
  'Communication Services': 'Telecomunicações',
};

const Map<String, String> fiSectorAliases = {
  'technology': 'Tecnologia',
  'finance': 'Financeiro',
  'healthcare': 'Saúde',
  'energy': 'Energia',
  'utilities': 'Utilidades Públicas',
  'consumer-discretionary': 'Consumo Cíclico',
  'consumer-staples': 'Consumo Básico',
  'industrials': 'Industrial',
  'materials': 'Materiais Básicos',
  'real-estate': 'Imobiliário',
  'telecommunications': 'Telecomunicações',
  'Miscellaneous': 'Outros',
  'Finance': 'Financeiro',
  'Technology Services': 'Tecnologia',
  'Electronic Technology': 'Tecnologia',
  'Producer Manufacturing': 'Industrial',
  'Industrial Services': 'Industrial',
  'Retail Trade': 'Consumo Cíclico',
  'Consumer Services': 'Consumo Cíclico',
  'Consumer Durables': 'Consumo Cíclico',
  'Process Industries': 'Materiais Básicos',
  'Non-Energy Minerals': 'Materiais Básicos',
  'Health Technology': 'Saúde',
  'Health Services': 'Saúde',
  'Consumer Non-Durables': 'Consumo Básico',
  'Commercial Services': 'Industrial',
  'Transportation': 'Industrial',
  'Energy Minerals': 'Energia',
  'Communications': 'Telecomunicações',
  'Distribution Services': 'Industrial',
};

const Map<String, int> fiSectorSeriesByLabel = {
  'Financeiro': 1,
  'Tecnologia': 2,
  'Energia': 3,
  'Consumo Cíclico': 4,
  'Saúde': 5,
  'Industrial': 6,
  'Imobiliário': 7,
  'Consumo Básico': 8,
  'Materiais Básicos': 9,
  'Utilidades Públicas': 10,
  'Telecomunicações': 11,
};

const Map<String, String> fiFixedIncomeKinds = {
  'cdb': 'CDB',
  'lci': 'LCI',
  'lca': 'LCA',
  'lc': 'LC',
  'cri': 'CRI',
  'cra': 'CRA',
  'tesouro_selic': 'Tesouro Selic',
  'tesouro_ipca': 'Tesouro IPCA+',
  'tesouro_pre': 'Tesouro Pré',
};

const Map<String, String> fiLiquidity = {
  'diaria': 'Liquidez diária',
  'no_vencimento': 'No vencimento',
};

const Map<String, FiCategory> fiExpenseCategories = {
  'moradia': FiCategory('Moradia', 1),
  'contas_da_casa': FiCategory('Contas da casa', 2),
  'mercado': FiCategory('Mercado', 3),
  'transporte': FiCategory('Transporte', 4),
  'saude': FiCategory('Saúde', 5),
  'educacao': FiCategory('Educação', 6),
  'lazer': FiCategory('Lazer', 7),
  'cuidados_pessoais': FiCategory('Cuidados pessoais', 8),
  'divida': FiCategory('Dívida', 9),
  'outros': FiCategory('Outros', 0),
};

const Map<String, FiCategory> fiIncomeCategories = {
  'salario': FiCategory('Salário', 1),
  'decimo_terceiro': FiCategory('13º salário', 2),
  'ferias': FiCategory('Férias', 3),
  'renda_variavel': FiCategory('Renda variável', 4),
  'provento': FiCategory('Provento', 5),
  'reembolso': FiCategory('Reembolso', 6),
  'outros': FiCategory('Outros', 0),
};

const Map<String, String> fiDebtKinds = {
  'rotativo_cartao': 'Rotativo do cartão',
  'cheque_especial': 'Cheque especial',
  'credito_pessoal': 'Crédito pessoal',
  'financiamento_imovel': 'Financiamento de imóvel',
  'financiamento_veiculo': 'Financiamento de veículo',
  'parcelamento': 'Parcelamento',
  'outros': 'Outros',
};

const Map<String, String> fiLedgerKinds = {
  'buy': 'Compra',
  'sell': 'Venda',
  'split': 'Desdobramento',
  'bonus': 'Bonificação',
  'transfer_in': 'Transferência de entrada',
  'transfer_out': 'Transferência de saída',
  'amortization': 'Amortização',
  'adjust': 'Declaração de posição',
};

const Map<String, String> fiLedgerKindExplanations = {
  'buy': 'Aumenta a quantidade e o custo. Entra no preço médio.',
  'sell': 'Reduz quantidade e custo, nunca a média — é a convenção brasileira. Apura ganho no mês.',
  'split': 'Multiplica a quantidade sem mexer no valor investido. Sem ele, o imposto sai errado.',
  'bonus': 'Ações recebidas sem desembolso. Aumentam a quantidade e diluem a média.',
  'transfer_in': 'Papel que chegou de outra corretora, com o custo que ele já tinha.',
  'transfer_out': 'Papel que saiu para outra corretora. Reduz a posição sem apurar ganho.',
  'amortization': 'Devolução de capital, comum em FII. Reduz o custo, não a quantidade.',
  'adjust': 'Ancora a linha do tempo: o que vier depois se aplica em cima do que você declarou.',
};

const Map<String, String> fiDividendKinds = {
  'dividendo': 'Dividendo',
  'jcp': 'JCP',
  'rendimento': 'Rendimento',
  'amortizacao': 'Amortização',
  'outro': 'Outro',
};

const Map<String, String> fiAccountData = {
  'positions': 'posições',
  'snapshots': 'fotografias do patrimônio',
  'goals': 'metas',
  'sector_goals': 'metas por setor',
  'preferences': 'preferências',
  'notified_opportunities': 'oportunidades já avisadas',
  'device_tokens': 'aparelhos que recebem aviso',
  'price_alerts': 'alertas de preço',
  'fixed_income_positions': 'aplicações de renda fixa',
  'dividends_received': 'proventos recebidos',
  'followed_suggestions': 'ativos seguidos',
  'transactions': 'livro-razão',
  'audit_log': 'registro de auditoria',
  'subscription': 'assinatura',
  'checkout_sessions': 'sessões de pagamento',
  'usage_counters': 'contadores de uso',
  'product_events': 'eventos de uso do produto',
  'revoked_tokens': 'sessões revogadas',
  'referral_code': 'seu código de indicação',
  'referrals_made': 'indicações feitas',
  'cash_entries': 'lançamentos do caixa',
  'cash_recurrences': 'lançamentos recorrentes',
  'debts': 'dívidas',
};

File: web/src/app/core/vocabulary.ts
// As bandas de régua e o vocabulário do produto, escritos à mão.
//
// Os limiares de score espelham `backend/app/analysis/score_ruler.py`, que é a fonte: mudar um
// limiar exige mudar o Python primeiro, e depois aqui e no espelho de `mobile/lib/core/`.

export interface FiCategoria {
  readonly label: string;
  readonly series: number;
  readonly icon: string;
}

export const fiCategorias: Readonly<Record<string, FiCategoria>> = {
  renda_fixa: { label: 'Renda Fixa', series: 1, icon: 'landmark' },
  acoes_br: { label: 'Ações BR', series: 2, icon: 'trending-up' },
  fiis: { label: 'FIIs', series: 3, icon: 'building-2' },
  bdrs: { label: 'BDRs', series: 5, icon: 'globe' },
  etfs: { label: 'ETFs', series: 8, icon: 'layers' },
  auto: { label: 'Automática', series: 0, icon: 'circle' },
};

export const fiCategoriaApelidos: Readonly<Record<string, string>> = {
  renda: 'renda_fixa',
  caixa: 'renda_fixa',
  trade: 'acoes_br',
};

export const fiTiposDeAtivo: Readonly<Record<string, { label: string; category: string }>> = {
  br_stock: { label: 'Ação BR', category: 'acoes_br' },
  bdr: { label: 'BDR', category: 'bdrs' },
  fii: { label: 'FII', category: 'fiis' },
  etf: { label: 'ETF', category: 'etfs' },
  renda_fixa: { label: 'Renda Fixa', category: 'renda_fixa' },
};

export const fiSetores: Readonly<Record<string, { label: string; series: number }>> = {
  'Financial Services': { label: 'Financeiro', series: 1 },
  Technology: { label: 'Tecnologia', series: 2 },
  Energy: { label: 'Energia', series: 3 },
  'Consumer Cyclical': { label: 'Consumo Cíclico', series: 4 },
  Healthcare: { label: 'Saúde', series: 5 },
  Industrials: { label: 'Industrial', series: 6 },
  'Real Estate': { label: 'Imobiliário', series: 7 },
  'Consumer Defensive': { label: 'Consumo Básico', series: 8 },
  'Basic Materials': { label: 'Materiais Básicos', series: 9 },
  Utilities: { label: 'Utilidades Públicas', series: 10 },
  'Communication Services': { label: 'Telecomunicações', series: 11 },
};

export const fiSetorApelidos: Readonly<Record<string, string>> = {
  technology: 'Tecnologia',
  finance: 'Financeiro',
  healthcare: 'Saúde',
  energy: 'Energia',
  utilities: 'Utilidades Públicas',
  'consumer-discretionary': 'Consumo Cíclico',
  'consumer-staples': 'Consumo Básico',
  industrials: 'Industrial',
  materials: 'Materiais Básicos',
  'real-estate': 'Imobiliário',
  telecommunications: 'Telecomunicações',
  Miscellaneous: 'Outros',
  Finance: 'Financeiro',
  'Technology Services': 'Tecnologia',
  'Electronic Technology': 'Tecnologia',
  'Producer Manufacturing': 'Industrial',
  'Industrial Services': 'Industrial',
  'Retail Trade': 'Consumo Cíclico',
  'Consumer Services': 'Consumo Cíclico',
  'Consumer Durables': 'Consumo Cíclico',
  'Process Industries': 'Materiais Básicos',
  'Non-Energy Minerals': 'Materiais Básicos',
  'Health Technology': 'Saúde',
  'Health Services': 'Saúde',
  'Consumer Non-Durables': 'Consumo Básico',
  'Commercial Services': 'Industrial',
  Transportation: 'Industrial',
  'Energy Minerals': 'Energia',
  Communications: 'Telecomunicações',
  'Distribution Services': 'Industrial',
};

export const fiSetorSeriePorRotulo: Readonly<Record<string, number>> = {
  Financeiro: 1,
  Tecnologia: 2,
  Energia: 3,
  'Consumo Cíclico': 4,
  Saúde: 5,
  Industrial: 6,
  Imobiliário: 7,
  'Consumo Básico': 8,
  'Materiais Básicos': 9,
  'Utilidades Públicas': 10,
  Telecomunicações: 11,
};

export const fiTiposDeRendaFixa: Readonly<Record<string, string>> = {
  cdb: 'CDB',
  lci: 'LCI',
  lca: 'LCA',
  lc: 'LC',
  cri: 'CRI',
  cra: 'CRA',
  tesouro_selic: 'Tesouro Selic',
  tesouro_ipca: 'Tesouro IPCA+',
  tesouro_pre: 'Tesouro Pré',
};

export const fiLiquidez: Readonly<Record<string, string>> = {
  diaria: 'Liquidez diária',
  no_vencimento: 'No vencimento',
};

export const fiCategoriasDeDespesa: Readonly<Record<string, FiCategoria>> = {
  moradia: { label: 'Moradia', series: 1, icon: 'house' },
  contas_da_casa: { label: 'Contas da casa', series: 2, icon: 'plug-zap' },
  mercado: { label: 'Mercado', series: 3, icon: 'shopping-cart' },
  transporte: { label: 'Transporte', series: 4, icon: 'compass' },
  saude: { label: 'Saúde', series: 5, icon: 'stethoscope' },
  educacao: { label: 'Educação', series: 6, icon: 'book-open' },
  lazer: { label: 'Lazer', series: 7, icon: 'shopping-bag' },
  cuidados_pessoais: { label: 'Cuidados pessoais', series: 8, icon: 'circle-user' },
  divida: { label: 'Dívida', series: 9, icon: 'receipt' },
  outros: { label: 'Outros', series: 0, icon: 'circle' },
};

export const fiCategoriasDeEntrada: Readonly<Record<string, FiCategoria>> = {
  salario: { label: 'Salário', series: 1, icon: 'wallet' },
  decimo_terceiro: { label: '13º salário', series: 2, icon: 'gift' },
  ferias: { label: 'Férias', series: 3, icon: 'sunrise' },
  renda_variavel: { label: 'Renda variável', series: 4, icon: 'chart-line' },
  provento: { label: 'Provento', series: 5, icon: 'coins' },
  reembolso: { label: 'Reembolso', series: 6, icon: 'circle-arrow-down' },
  outros: { label: 'Outros', series: 0, icon: 'circle' },
};

export const fiTiposDeDivida: Readonly<Record<string, string>> = {
  rotativo_cartao: 'Rotativo do cartão',
  cheque_especial: 'Cheque especial',
  credito_pessoal: 'Crédito pessoal',
  financiamento_imovel: 'Financiamento de imóvel',
  financiamento_veiculo: 'Financiamento de veículo',
  parcelamento: 'Parcelamento',
  outros: 'Outros',
};

export const fiClasseTextoDaSerie: Readonly<Record<number, string>> = {
  0: 'text-series-other',
  1: 'text-series-1',
  2: 'text-series-2',
  3: 'text-series-3',
  4: 'text-series-4',
  5: 'text-series-5',
  6: 'text-series-6',
  7: 'text-series-7',
  8: 'text-series-8',
  9: 'text-series-9',
};

export const fiClasseFundoDaSerie: Readonly<Record<number, string>> = {
  0: 'bg-series-other',
  1: 'bg-series-1',
  2: 'bg-series-2',
  3: 'bg-series-3',
  4: 'bg-series-4',
  5: 'bg-series-5',
  6: 'bg-series-6',
  7: 'bg-series-7',
  8: 'bg-series-8',
  9: 'bg-series-9',
};

export const fiClasseBarraDaSerie: Readonly<Record<number, string>> = {
  0: 'bg-series-other',
  1: 'bg-series-1',
  2: 'bg-series-2',
  3: 'bg-series-3',
  4: 'bg-series-4',
  5: 'bg-series-5',
  6: 'bg-series-6',
  7: 'bg-series-7',
  8: 'bg-series-8',
  9: 'bg-series-9',
};

export const fiClasseChipDaSerie: Readonly<Record<number, string>> = {
  0: 'bg-series-other/15',
  1: 'bg-series-1/15',
  2: 'bg-series-2/15',
  3: 'bg-series-3/15',
  4: 'bg-series-4/15',
  5: 'bg-series-5/15',
  6: 'bg-series-6/15',
  7: 'bg-series-7/15',
  8: 'bg-series-8/15',
  9: 'bg-series-9/15',
};

export const fiClasseBordaDaSerie: Readonly<Record<number, string>> = {
  0: 'border-series-other/30',
  1: 'border-series-1/30',
  2: 'border-series-2/30',
  3: 'border-series-3/30',
  4: 'border-series-4/30',
  5: 'border-series-5/30',
  6: 'border-series-6/30',
  7: 'border-series-7/30',
  8: 'border-series-8/30',
  9: 'border-series-9/30',
};
